orchagent/neighorch.h

```diff
@@ -16,23 +20,48 @@ struct NeighborEntry
     }
 };
 
+struct NextHopEntry
+{
+    sai_object_id_t     next_hop_id;    // next hop id or next hop group id
+    int                 ref_count;      // reference count
+};
 
 /* NeighborTable: NeighborEntry, neighbor MAC address */
 typedef map<NeighborEntry, MacAddress> NeighborTable;
+/* NextHopTable: next hop IP address(es), NextHopEntry */
+typedef map<IpAddresses, NextHopEntry> NextHopTable;
 
 class NeighOrch : public Orch
 {
 public:
-    NeighOrch(DBConnector *db, string tableName, PortsOrch *portsOrch, RouteOrch *routeOrch) :
-        Orch(db, tableName), m_portsOrch(portsOrch), m_routeOrch(routeOrch) {};
+    NeighOrch(DBConnector *db, string tableName, PortsOrch *portsOrch) :
+        Orch(db, tableName),
+        m_portsOrch(portsOrch),
+        m_nextHopGroupCount(0) {};
+
+    bool contains(IpAddress);
+    bool contains(IpAddresses);
+
+    bool addNextHopGroup(IpAddresses);
+    bool removeNextHopGroup(IpAddresses);
```

Comment:
The next hop group goes with next hop as it is just a bunch of neighbors forming a next hop group. As and the m_syncdNextHops data structure is in this class, it makes sense to move addNextHopGroup/removeNextHopGroup method to this class. So all the next hop (group) concepts are here.